src/upkeep.rs

```diff
@@ -205,7 +205,13 @@ pub async fn do_upkeep(
 
                     match delivery {
                         Ok(_) => Ok(activation.id),
-                        Err(err) => Err(err),
+                        Err((err, _msg)) => {
+                            error!(
+                                "deadletter.publish.failure: {}, message: {:?}",
+                                err, payload
+                            );
+                            Err(err)
```

Comment:
Should we be returning `Some(activation.id)` so that the activation will be deleted?